File: source/cpp/27. Remove Element.cpp
#include <vector>


class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        if (nums.size() == 0) return 0;

        while (true) {
            auto it = std::find(nums.begin(), nums.end(), val);
            if (it == nums.end()) break;
            nums.erase(it);
        }

        return nums.size();
    }
};

File: source/python/27. Remove Element.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        while True:
            try:
                nums.remove(val)
            except ValueError:
                break

        return len(nums)

    # def removeElement(self, nums: list[int], val: int) -> int:
    #     if not nums:
    #         return 0

    #     insert_index = 0
    #     for item in nums:
    #         if item != val:
    #             nums[insert_index] = item
    #             insert_index += 1

    #     return insert_index


def test():
    s = Solution()

    nums = [0, 1, 2, 2, 3, 0, 4, 2]
    val = 2
    assert s.removeElement(nums, val) == 5

    nums = [3, 2, 2, 3]
    val = 3
    assert s.removeElement(nums, val) == 2

    print("All tests passed!")


if __name__ == "__main__":
    test()
